File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class PersonalRecord(BaseModel):
    exercise: str
    weight: float
    reps: int
    estimated_1rm: float


class AnalyticsSummary(BaseModel):
    total_workouts: int
    workouts_this_week: int
    weekly_volume: float
    best_estimated_1rm: float | None
    personal_records: list[PersonalRecord]

File: frontend/src/types/analytics.ts
export interface PersonalRecord {
  exercise: string;
  weight: number;
  reps: number;
  estimated_1rm: number;
}


export interface AnalyticsSummary {
  total_workouts: number;
  workouts_this_week: number;
  weekly_volume: number;
  best_estimated_1rm: number | null;
  personal_records: PersonalRecord[];
}